source/common/ssl/context_config_impl.cc

```diff
@@ -139,17 +169,21 @@ ServerContextConfigImpl::ServerContextConfigImpl(
         return ret;
       }()) {
   // TODO(PiotrSikora): Support multiple TLS certificates.
-  if (config.common_tls_context().tls_certificates().size() != 1) {
+  if (config.common_tls_context().tls_certificates().size() != 1 &&
```

Comment:
I don't think this is right. You need there to be at least 1 cert across both sources, right? Assuming so, please add a test that should fail with this code and then fix it.

Reply:
There need to be exactly 1 cert.

Reply:
Modified to check the sum is 1